test/helpers/gem_helpers.rb

```diff
@@ -17,7 +17,7 @@ def build_gem(name, version, summary = "Gemcutter", platform = "ruby", &block)
     build_gemspec(new_gemspec(name, version, summary, platform, &block))
   end
 
-  def new_gemspec(name, version, summary, platform, &block)
+  def new_gemspec(name, version, summary, platform, metadata = nil, &block)
```

Comment:
My idea behind this was to remove responsibility for this in `app/models/linkset.rb` and remove `meta = spec.metadata || {}` line. I think this is best place to take responsibility.